src/Http/Params.php

```diff
@@ -38,7 +38,7 @@ public function __construct(array|string|null $params)
 	 */
 	public static function extract(string|array|null $path = null): array
 	{
-		if (empty($path) === true) {
+		if ($path === null || $path === '' || $path === []) {
```

Comment:
I'm sorry, but I don't like this syntax 😅 It may use for a single type as you did, but writing it separately for 3 different types is not readable at all. IMHO `empty($path) === true` is simple and clear 🙂

Reply:
It's not about speed/performance and it's the contrary in terms of readability: `empty()` has worse readability as it obfuscates what it does. As the article says:

> - it tests both for the existance and the non-falsy value and as a reader the writers intent is not clear.
>- it does not communicate what type a variable has, seeing empty($value) does not narrow down what value is enough.
>- It hides errors caused by typos in the variable name passed to empty.

That's why many PHP developers try to ban `empty()` from there code. And I think we should try as much as we can when revising and writing new code. This should not be about taste - I agree the short `empty()` looks better - but what helps us in the long term to write more reliable, less error prone code.